sigstore-java/src/main/java/dev/sigstore/bundle/BundleFactory.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.bundle;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import dev.sigstore.KeylessSigningResult;
+import dev.sigstore.proto.bundle.v1.Bundle;
+import dev.sigstore.proto.bundle.v1.VerificationMaterial;
+import dev.sigstore.proto.common.v1.*;
+import dev.sigstore.proto.rekor.v1.*;
+import dev.sigstore.rekor.client.RekorEntry;
+import java.security.cert.CertificateEncodingException;
+import java.util.Base64;
+import java.util.stream.Collectors;
+
+/**
+ * Generates Sigstore Bundle.
+ */
+public class BundleFactory {
+  private static final JsonFormat.Printer JSON_PRINTER = JsonFormat.printer();
+
+  /**
+   * Generates Sigstore Bundle from {@link KeylessSigningResult}.
+   * @param signingResult Keyless signing result.
+   * @return Sigstore Bundle in JSON format
+   * @throws IllegalArgumentException in case
+   */
+  public static String createBundle(KeylessSigningResult signingResult) {
+    Bundle bundle = createBundleProto(signingResult);
+    try {
+      return JSON_PRINTER.print(bundle);
```

Comment:
Do we just produce `String` or would it be helpful to return `Bundle` as well?

Reply:
I think returning Bundle would be nice. The client doesn't necessarily *need* to use Json, though it would be odd if we didn't standardize on something.

Presumably KeylessVerifier could verify on the bundle type itself (rather than the individual components like it does now).

Reply:
> Presumably KeylessVerifier could verify on the bundle type itself

It should probably verify on the bundle `byte[]` (e.g. UTF-8 that stores the JSON).

Reply:
Exposing protobuf types on the public API would require promoting `protobuf-java` to `api(...)`, so the users would get `protobuf-java` in their autocomplete menus even in case they do not use protobuf.

For now I expose only JSON as String, and I guess we can expose protobuf type if needed later.